packages/odyssey-react-mui/src/labs/DataComponents/StackContent.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  MRT_Row,
+  MRT_RowData,
+  MRT_RowSelectionState,
+  MRT_TableInstance,
+} from "material-react-table";
+
+import { Box } from "../../Box";
+import { Checkbox as MuiCheckbox } from "@mui/material";
+import { CircularProgress } from "../../CircularProgress";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import { RowActions } from "./RowActions";
+import { StackCard } from "./StackCard";
+import { StackLayout, StackProps, UniversalProps } from "./componentTypes";
+
+export type StackContentProps = {
+  currentLayout: StackLayout;
+  data: MRT_RowData[];
+  draggingRow?: MRT_Row<MRT_RowData> | null;
+  emptyState: ReactNode;
+  getRowId: UniversalProps["getRowId"];
+  hasRowReordering: UniversalProps["hasRowReordering"];
+  hasRowSelection: UniversalProps["hasRowSelection"];
+  isEmpty?: boolean;
+  isLoading: boolean;
+  isNoResults?: boolean;
+  isRowReorderingDisabled?: boolean;
+  onReorderRows: UniversalProps["onReorderRows"];
+  pagination: { pageIndex: number; pageSize: number };
+  rowReorderingUtilities: {
+    dragHandleStyles: CSSObject;
+    dragHandleText: {
+      title: string;
+      "aria-label": string;
+    };
+    draggableTableBodyRowClassName: ({
+      currentRowId,
+      draggingRowId,
+      hoveredRowId,
+    }: {
+      currentRowId: string;
+      draggingRowId?: string;
+      hoveredRowId?: string;
+    }) => string | undefined;
+    handleDragHandleKeyDown: ({
+      table,
+      row,
+      event,
+    }: {
+      table: MRT_TableInstance<MRT_RowData>;
+      row: MRT_Row<MRT_RowData>;
+      event: React.KeyboardEvent<HTMLButtonElement>;
+    }) => void;
+    handleDragHandleOnDragCapture: (
+      table: MRT_TableInstance<MRT_RowData>,
+    ) => void;
+    handleDragHandleOnDragEnd: (table: MRT_TableInstance<MRT_RowData>) => void;
+    resetDraggingAndHoveredRow: (table: MRT_TableInstance<MRT_RowData>) => void;
+    updateRowOrder: ({
+      rowId,
+      newRowIndex,
+    }: {
+      rowId: string;
+      newRowIndex: number;
+    }) => void;
+  };
+  rowSelection: MRT_RowSelectionState;
+  setRowSelection: Dispatch<SetStateAction<MRT_RowSelectionState>>;
+  stackOptions: StackProps;
+  totalRows: UniversalProps["totalRows"];
+};
+
+const StackContent = ({
+  currentLayout,
+  data,
+  emptyState,
+  hasRowReordering,
+  hasRowSelection,
+  isEmpty,
+  isLoading,
+  isNoResults,
+  isRowReorderingDisabled,
+  onReorderRows,
+  pagination,
+  rowReorderingUtilities,
+  rowSelection,
+  setRowSelection,
+  stackOptions,
+  totalRows,
+}: StackContentProps) => {
+  const handleRowSelectionChange = useCallback(
+    (row: MRT_RowData) => {
+      setRowSelection((prev) => {
+        const newSelection = { ...prev };
+        if (newSelection[row.id]) {
+          delete newSelection[row.id];
+        } else {
+          newSelection[row.id] = true;
+        }
+        return newSelection;
+      });
```

Comment:
This is how I'd probably write this, but that doesn't mean you have to change anything:
```suggestion
      setRowSelection((rowSelection) => (
        Object.fromEntries(
          row.id in rowSelection
          ? (
            Object.entries(
              currentRowSelection
            )
            .filter(([key]) => (
              key === row.id
            ))
          )
          : (
            Object.entries(
              rowSelection
            )
            .concat([
              row.id,
              true,
            ])
          )
        )
      ));
```

I prefer avoiding `const` and `delete` if I can as they're imperative rather than declarative. Doesn't mean the declarative one is better or easier to read or even more performant; it's just a different way of writing things.

Either way, we should probably rename these variables at least:
```suggestion
      setRowSelection((rowSelection) => {
        const nextRowSelection = { ...rowSelection };
        if (nextRowSelection[row.id]) {
          delete nextRowSelection[row.id];
        } else {
          nextRowSelection[row.id] = true;
        }
        return nextRowSelection;
      });
```

I don't like the word `prev` because it doesn't tell you anything about what it is.